query_execution/WorkOrdersContainer.hpp

```diff
@@ -311,16 +311,17 @@ class WorkOrdersContainer {
     }
 
     inline void addWorkOrder(WorkOrder *workorder) {
-      workorders_.push(workorder);
+      workorders_.emplace(workorder);
```

Comment:
Oh, I got the point: if an exception happens during the `emplace`, there will be memory leak for newly created `WorkOrder`. (Btw, the original version with `push_back` should have the same memory leak issue).

And as suggested by the same post, we could avoid the problem using the following code:

```c++
WorkOrder *workorder = new WorkOrder;

std::list<std::unique_ptr<WorkOrder>> workorders_;
workorders_.emplace_back(std::unique_ptr<WorkOrder>(workorder));
```